packages/grid_client/src/high_level/machine.ts

```diff
@@ -76,6 +76,15 @@ class VMHL extends HighLevelBase {
       throw new GridClientErrors.Nodes.UnavailableNodeError(
         `Node ${nodeId} is not available for user with twinId: ${twinId}, maybe it's rented by another user or node is dedicated. use capacity planning with availableFor option.`,
       );
+    } else {
+      // If Available for twinId (dedicated), check it's not in grace period
+      const NodeInfo = await this.nodes.getNode(nodeId);
+      const contract = await this.config.tfclient.contracts.get({ id: NodeInfo.rentContractId });
+      if (contract.state.gracePeriod) {
+        throw new GridClientErrors.Nodes.UnavailableNodeError(
+          `Can't deploy on node: ${nodeId}, its contract in grace period.`,
+        );
+      }
```

Comment:
why don't we use `validateRentContract` instead of this?

Reply:
`validateRentContract` in playground not grid_client